x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java

```diff
@@ -122,6 +125,35 @@ public class InternalUsers {
         )
     );
 
+    /**
+     * Internal user that manages DLM. Has all indices permissions to perform DLM runtime tasks.
+     */
+    public static final InternalUser DLM_USER = new InternalUser(
+        UsernamesField.DLM_NAME,
+        new RoleDescriptor(
+            UsernamesField.DLM_ROLE,
+            new String[] {},
+            new RoleDescriptor.IndicesPrivileges[] {
+                RoleDescriptor.IndicesPrivileges.builder()
+                    // There are no active plans to manage the security index or async search index with DLM, so excluding them here
+                    .indices("/@&~(\\.security.*)&~(\\.async-search.*)/")
+                    .privileges(
+                        "delete_index",
+                        RolloverAction.NAME,
+                        ForceMergeAction.NAME + "*",
+                        // indices stats is used by rollover, so we need to grant it here
+                        IndicesStatsAction.NAME + "*"
+                    )
+                    .allowRestrictedIndices(true)
```

Comment:
Why is this needed ? I am not aware of too many system data streams, but for those, if we want to use DLM to manage them, I would prefer a different approach to address at a later time. (that approach would likely be a different custom user with explicit privileges for that exact system index and a custom API (and maybe even custom privilges) to update/change the management for the system index/data streams).

Reply:
This is based on discussions with @masseyke  and @andreidan:

In short, we do plan to use DLM for managing system data streams. In particular for stateless, DLM will be the only way to do this. Currently, the only system data stream in ES is `.fleet-actions-results`. I will follow up with the Fleet team to check if this is something we plan to support for stateless or not. 

Currently, the DLM implementation supports interacting with system data streams via the lifecycle APIs, but restricts access based on product origin headers. There's an explicit test for this behavior: https://github.com/elastic/elasticsearch/blob/main/modules/dlm/src/internalClusterTest/java/org/elasticsearch/dlm/CrudDataLifecycleSystemDataStreamIT.java. The other place where DLM comes into play for system data streams is index templates. I've pinged core-infra to confirm this as well, but my current understanding is that system data streams define their own, hard-coded templates and as such there is no way for end-users to mess with DLM for these, via the templates APIs or by any other means.

We need to discuss whether we want to move away from this in favor of a custom API, and align on our general approach to configuring DLM for system data streams. I think that goes beyond the scope of the current PR though.

In terms of privileges for the runtime DLM user: we can, in principle, tweak the DLM service implementation to pick a different origin based on whether it's running an action against a system data stream or not. If we want to use specific users for specific system streams to carry out runtime actions, this could get fairly complex. We'd have to switch the action origin (e.g., by using a different origin-setting client) based on information available on the data stream (e.g., name and isSystem). Alternatively, we could go for a different, bigger refactor to separate system from non-system data stream DLM runtime handling. 

I agree it would be nice to split out privileges across dedicated users, but I'm wondering if the benefits justify the overhead in this particular case: if a data stream has DLM configured, it _will_ be processed by the DLM service at runtime (using one internal user or another). From a security perspective, I think it's more important to come up with a way to restrict access to configuring DLM for system data streams, and less so around the runtime user. Viewed another way, I think the `_dlm` runtime user can have the privileges necessary to orchestrate all DLM service runtime actions (including on specific system data streams if needed), and it's a separate concern to ensure that a user _configuring_ DLM has the requisite privileges to do so.  

As such, I wonder if we should grant restricted indices access to `_dlm` but scope it down to specific system data streams (in this case, we would only start with `.fleet-actions-results`, if applicable). Happy to discuss!

Reply:
> As such, I wonder if we should grant restricted indices access to _dlm but scope it down to specific system data streams (in this case, we would only start with .fleet-actions-results, if applicable).

That would make me more comfortable. As you point out I don't think there is an immediate problem giving this internal user that privilege. The concern is more about generally adoption of least privilege and when in doubt leave it out. IMO if we know the exact names of the system data streams we should explicitly grant access to that name. However, I would still prefer to leave it out if it is not planned to be used and we can introduce the role addition at the same time.

> system data streams define their own, hard-coded templates and as such there is no way for end-users to mess with DLM for these

That is my understanding too. However, I think we should still adhere to least privs.

Reply:
Discussed on Slack:

We will go ahead with specifying a specific reserved data stream name (`.fleet-actions-responses`) to start with, and align on configuration time privilege checks separately from this PR. Will update the code accordingly, thanks for raising this!